core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/EvaluationStrategyFactory.java

```diff
@@ -0,0 +1,17 @@
+/**
+ * Copyright (c) 2015 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+package org.eclipse.rdf4j.query.algebra.evaluation;
+
+import org.eclipse.rdf4j.query.Dataset;
+
+/**
+ * Factory for {@link EvaluationStrategy}s.
+ */
+public interface EvaluationStrategyFactory {
+	EvaluationStrategy createEvaluationStrategy(Dataset dataset, TripleSource tripleSource);
```

Comment:
This needs javadoc. In particular, a reference to calling EvaluationStrategies.register as a recommendation to implementors before returning the EvaluationStrategy from this method if that is desired.